pkg/lib/ownerutil/util.go

```diff
@@ -257,17 +216,24 @@ func IsOwnedByKindLabel(object metav1.Object, ownerKind string) bool {
 	return object.GetLabels()[OwnerKind] == ownerKind
 }
 
-// AdoptableLabels determines if an OLM managed resource is adoptable based on its owner labels
-// Generally used for cross-namespace ownership and for Cluster -> Namespace scope
-func AdoptableLabels(target Owner, labels map[string]string) bool {
+// AdoptableLabels determines if an OLM managed resource is adoptable by any of the given targets based on its owner labels.
+// The checkName perimeter enables an additional check for name equality with the `olm.owner` label.
+// Generally used for cross-namespace ownership and for Cluster -> Namespace scope.
+func AdoptableLabels(labels map[string]string, checkName bool, targets ...Owner) bool {
```

Comment:
What scenario would you not want to check the name?

Reply:
That's a super good point. This would be useful in situations where all we care about is that the resource was OLM generated or explicitly set to be managed by OLM. The thought process is a carry-over from `OwnerReferences` - we didn't want to adopt a resource that wasn't managed by OLM since we didn't want it to be deleted when the owner was.

Reply:
Looks like there's no place where `false` is being passed in though. Maybe it's better to remove it altogether.